test/test_team.py

```diff
@@ -438,6 +438,42 @@ def rough_bot(bot, state):
     assert state['errors'] == [{}, {}]
     assert state['fatal_errors'] == [[], []]
 
+def test_bot_graph_is_half_mutable():
+    # Test that a bot can change the weights of the graph
+    # Changing the weights affects the future self of the bot
+    # Changing the weights does not affect the other bot
+    layout = """
+    ########
+    #  ax  #
+    #  by  #
+    #......#
+    ########
+    """
+    def blue(bot, state):
+        if bot.turn == 0 and bot.round == 1:
+            assert bot.graph[1, 1][1, 2].get('weight') is None
+            bot.graph[1, 1][1, 2]['weight'] = 1
+        else:
+            assert bot.graph[1, 1][1, 2].get('weight') == 1
```

Comment:
wait: from this line it seems that the changed weights are indeed visible to the other bot?

Reply:
does changing the weights affect the computation of the shortest path? This could be the source of very interesting bugs :-)